### ISSUE TITLE
Brainstorm

### ISSUE THREAD

**Issue by hermanbanken** [WISVCH/website#1]
We zijn aan het brainstormen over een nieuwe website. Er zijn heel veel mogelijkheden, zoals integratie met bestaande CH systemen, leuke features voor leden en API's voor app's etc.

Drupal wordt door de huidige lichting W3Cie niet helemaal ideaal bevonden. Brainstormen over een nieuwe basis kan ook hier. Willen we een bestaand CMS, iets hips nieuws, zelf custom gemaakt, etc?

**Comment by hermanbanken** [comment]
Huidige features zijn:
- Nieuwsberichten
- Pagina's
- Evenementen / kalendar
- Bedrijven / bedrijfsprofielen
- Advertenties
- Vacatures
- TV items
- Uploads (CHoice)
- Taxonomieen (voor CHoice en commissies, besturen, etc.)

**Comment by sanderploegsma** [comment]
Het uiteindelijke idee om de website op een API te laten draaien leidt tot de discussie om een gecentraliseerde service te creëren die (uiteindelijk) met alle diensten van CH kan werken. Deze kan vervolgens worden gebruikt door o.a. de website en de TV's, evenals alle andere dingen die bijvoorbeeld andere CH'ers willen toevoegen.

De grote uitdaging hierin zit in de beveiliging en op welke manier welke mensen welke data (WWW... :wink: ) kunnen gebruiken. Koppelingen met o.a. LDAP en dienstsysteem zijn daarom dingen die waarschijnlijk met de hulp en ervaring van verschillende beheerders gerealiseerd moeten worden.

**Comment by sanderploegsma** [comment]
Voor nu: met een proof of concept (poc) voor de website, inclusief een aparte service voor de api kunnen we kijken wat de mogelijkheden zijn en of deze bevallen. Voordelen hiervan zijn:

- De implementatie van de website zelf is veel simpeler en kan makkelijker onderhouden, aangepast of zelfs helemaal vernieuwd worden. Details zoals frameworks en dergelijke zouden hierbij niet uit moeten maken.
- Wanneer de api-service gekoppeld wordt aan andere systemen, heeft dat geen directe gevolgen voor de website.
- Iedereen vindt het leuk.

Het achterliggende idee is vrij duidelijk: onze website, infrastructuur en andere voorzieningen zouden *hackable* moeten zijn, zodat iedereen zijn random beunprojecten op zou kunnen zetten. Genoemde ideeën zijn dingen als:

* Een 'welk lid drinkt het meeste koffie'-scoreboard en andere implementaties die gebruik maken van een 'CH-pas'. 
* Taggen van CH-leden in de FlitCie foto's
* Een 'chat' maken rondom CHoice om tentamens met elkaar te kunnen bespreken.

**Comment by sanderploegsma** [comment]
En nog even voor de 'duidelijkheid':

![screen shot 2015-02-10 at 21 16 45](https://cloud.githubusercontent.com/assets/1564254/6135579/2e18cb6e-b16a-11e4-878e-ce6b6e32d5fa.png)

**Comment by hermanbanken** [comment]
Voor mijn idee met Meteor heb ik even een starter template geopend en daar wat entities aan toegevoegd. Deze starter staat in de branch [experiments/meteor](https://github.com/WISVCH/website/tree/experiment/meteor).

Het nadeel van wat ik me nu bedenk:
- Meteor werkt idealiter met een MongoDB, met volledige access. Een API strookt dus niet helemaal met dit idee. We hebben ook niet echt de RealTime capabiliteiten nodig voor de basic site natuurlijk. Al is het voor chat wel nice.

**Comment by sanderploegsma** [comment]
Heb even zitten spelen met Jersey, een framework voor REST applicaties, te vinden in de branch experiments/java-jersey. Even een project opzetten met een tomcat server en maven en je kan gaan!

Voorbeeld voor een call: ```curl -i http://localhost:8080/api/item/1``` geeft als response ```{"name":"New item","id":1}```

Was makkelijker dan ik dacht :smile_cat:

**Comment by MikeN123** [comment]
Pak dan meteen Dropwizard, http://dropwizard.io/

Eerlijkgezegd vind ik het een kutidee om alles maar zelf te bouwen, dat hebben we namelijk al 3x eerder gedaan en zorgt altijd voor iets wat de origineel bouwer heel awesome vindt, maar wat de rest van de wereld niet snapt, maar jullie zoeken het maar uit :hear_no_evil:

**Comment by evpaassen** [comment]
> Eerlijkgezegd vind ik het een kutidee om alles maar zelf te bouwen, dat hebben we namelijk al 3x eerder gedaan en zorgt altijd voor iets wat de origineel bouwer heel awesome vindt, maar wat de rest van de wereld niet snapt, maar jullie zoeken het maar uit :hear_no_evil:

Dat. Als er een bestaande site is op basis van kant en klare software dan is er altijd wel iemand te vinden die iets customs wil bouwen omdat het flexibeler kan en zo weet ik ook zeker dat er binnen twee jaar wel weer iemand opstaat die de nieuwe custom oplossing eruit wil mikken omdat het niet onderhoudbaar is voor een vereniging.

> De implementatie van de website zelf is veel simpeler en kan makkelijker onderhouden, aangepast of zelfs helemaal vernieuwd worden. Details zoals frameworks en dergelijke zouden hierbij niet uit moeten maken.

Ik kan me ook niet voorstellen hoe het onderhouden van een custom site ooit makkelijker/minder werk kan zijn dan eens in de zoveel tijd `drush up` typen.

**Comment by hermanbanken** [comment]
Het is de bedoeling om meer mensen bij de website te betrekken. We zijn een wiskunde en informatica vereniging en het is raar dat eigenlijk niemand aan de website werkt. Er zijn hele gave dingen mogelijk, maar je wilt niet iedereen deploy rechten hoeven geven. Daarom willen we de site open-sourcen, een API en een dev API hebben en hippe technologie gebruiken.

Drupal is niet iets waar je sjaars en twaars enthousiast voor maakt. Sorry @evpaassen.

**Comment by evpaassen** [comment]
Sjaars kunnen sowieso beter eerst even wat studiepunten halen.

**Comment by hermanbanken** [comment]
Uit eigen ervaring kan ik vertellen dat je als masterstudent meer tijd per studiepunt kwijt bent dan als sjaars. Maar hier gaat de discussie niet over.

De mensen die bij de W3Cie meeting waren dinsdag (11-02) waren allemaal niet blij of bekend met Drupal.

**Comment by MikeN123** [comment]
De technologie die hip is wisselt iedere 3 maanden, ik neem aan dat we dan ook de website iedere 3 maanden opnieuw bouwen?

**Comment by tomverhoeff** [comment]
ja!

**Comment by MikeN123** [comment]
Ja, alles in Windows 10 Developer Preview 3 Universal Apps RC3 Update

**Comment by jabbink** [comment]
![image](https://cloud.githubusercontent.com/assets/2461972/6390525/53f2fbd0-bdad-11e4-80d1-da76176ccb4c.png)
